### DISCUSSION TITLE
Rotation with a mouse drag

### DISCUSSION THREAD

**Discussion by paireks** [dotnet/Silk.NET#1210] [Q&A] [answered]
Hey guys!

I really apprieciate the example files for OpenGL and Imgui, they explain a lot. One thing I try to figure out, is making it possible to rotate 3d models using mouse dragging. In example files there is some sort of a fly mode, which is really cool and helpful, but at the same time for CAD related things rotating it with mouse dragging is much more popular. What I mean by this is something like e.g. this: https://stemkoski.github.io/Three.js/Labeled-Geometry.html

I understand library itself is pretty low-level, but maybe any of you have any examples or code samples where you rotate it not by flying with arrows and mouse, but with just dragging the mouse? :)

**Comment by desiwko** [comment, marked as answer]
This functionality is a pretty simple algorithm non-specific to any particular code.  Simplified algorithm:

1. Capture the initial mouse down coordinate
2. Track the mouse move coordinate while mouse down
3. Calculate the vector between the two points, normalize if/as desired
4. Create a rotation matrix from calculated vector
5. Apply rotation matrix to object
6. On mouse up, break out of the tracking loop

**Reply by paireks** [reply]
Then I think calculating step 4. (vector to rotation matrix) is what I'm looking for :) I will try to find it on web, thank you

**Comment by paireks** [comment]
Just a small note that there is a repo that does that: https://github.com/batu92k/Mouse-Orbit
